### PR TITLE
using index.cds

### PR DESCRIPTION
**Pull request by ThePlenkov**
Hi! 

Following this [guide ](https://cap.cloud.sap/docs/guides/extensibility/composition#index-cds) I'd like to add also index.cds as we have in a [bookshop ](https://github.com/SAP-samples/cloud-cap-samples/blob/main/bookshop/index.cds)

I'm using this package as a dependency using git submodules but I'd like to resolve it via package name rather than absolute path.

### CODE REVIEW

**Comment by ThePlenkov** [comment]
@stewsk Hi, can we merge this PR? thanks!

File: index.cds [hunk 1]
```diff
@@ -0,0 +1,2 @@
+namespace sap.capire.sflight; //> important for reflection
+using from './db/schema';
```
Comment (stewsk) [inline index.cds:2]: wouldn't you also need `using`s for the files in `srv`?

Comment (ThePlenkov) [reply index.cds:2]: I don't think so. According to this [guide ](https://cap.cloud.sap/docs/guides/extensibility/composition#index-cds) it seems like just enough to have index.cds as the entry point

Comment (stewsk) [reply index.cds:2]: But the example given in the [guide](https://cap.cloud.sap/docs/guides/extensibility/composition#index-cds) _does_ have `using from` also for the service files -> ?

File: package.json [hunk 2]
```diff
@@ -121,5 +121,6 @@
   "sapux": [
     "app/travel_processor",
     "app/travel_analytics"
-  ]
+  ],
+  "workspaces":["."]
```
Comment (danjoa) [inline package.json:125]:
```suggestion
```
That should not be required _within_ the project itself.

File: srv/travel-service.cds [hunk 3]
```diff
@@ -1,4 +1,4 @@
-using { sap.fe.cap.travel as my } from '../db/schema';
+using { sap.fe.cap.travel as my } from '@capire/sflight';
```
Comment (danjoa) [inline srv/travel-service.cds:1]:
```suggestion
using { sap.fe.cap.travel as my } from '../db/schema';
```
While we can do that _within_ the project itself, that is not reasonable.

File: index.cds (hunk 1 above)
Comment (danjoa) [reply index.cds:2]: The idea of the `index.cds` is that it controls what _external_ users of this package should have access to via `using {...} from '@capire/sflight'`. If we'd only want them to reuse our domain model, then we could keep it like that. If we'd also allow them to reuse our services (locally or remotely), then we should include them in the `index.cds`, ...

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -11,7 +11,14 @@
...
 	// Use 'postCreateCommand' to run commands after the container is created.
-	"postCreateCommand": "npm install"
+	"postCreateCommand": "npm install",
+	"customizations": {
+		"vscode": {
+			"extensions": [
+				"SAPSE.vscode-cds"
+			]
+		}
+	}
```
Comment (danjoa) [inline .devcontainer/devcontainer.json:21]: That change is not related to the intent of this PR, is it?

**Review by danjoa — CHANGES_REQUESTED** [review]
